src/daft-logical-plan/src/optimization/rules/push_down_aggregation.rs

```diff
@@ -45,11 +47,24 @@ impl OptimizerRule for PushDownAggregation {
                                 SourceInfo::Physical(external_info) => {
                                     let scan_op = external_info.scan_state.get_scan_op().0.clone();
 
-                                    // Check if scan operation supports aggregation pushdown and specific count mode
-                                    if scan_op.supports_count_pushdown()
-                                        && is_count_mode_supported(count_mode)
-                                        && external_info.pushdowns.filters.is_none()
+                                    // Enhanced check: support filter+count pushdown
+                                    let is_remaining_filters = if let Some(supports_pushdown) =
+                                        scan_op.as_pushdown_filter()
+                                        && self.strict_pushdown
                                     {
+                                        let (_pushed_filters, post_filters) = supports_pushdown
+                                            .push_filters(
+                                                external_info.pushdowns.filters.as_slice(),
+                                            );
+                                        post_filters.is_empty()
+                                    } else {
+                                        external_info.pushdowns.filters.is_none()
+                                    };
+                                    let can_pushdown = scan_op.supports_count_pushdown()
+                                        && is_count_mode_supported(count_mode)
+                                        && is_remaining_filters;
+
+                                    if can_pushdown {
```

Comment:
Let's add some tests on the Rust side for this new logic

Reply:
Thank you. I have completed it